Makefile

```diff
@@ -62,12 +67,14 @@ test: manifests generate fmt vet envtest ## Run tests.
 ##@ Build
 
 .PHONY: build
-build: generate fmt vet ## Build manager binary.
-	go build -o bin/manager main.go
+build: $(addprefix bin,/$(BINARIES)) ## Build manager binary.
+
+bin/%: main.go generate fmt vet FORCE
```

Comment:
Do we need main.go as dependency here? we're forcing this rule anyway so a rebuild will happen whether main.go has changed or not.

Reply:
It is for the `$<`

Reply:
Hmm. These makefile changes seem to introduce unneeded complexity. This repo is likely only ever to produce a single image and binary. I agree the current makefile (generated by kubebuilder) has some strangeness, but I don't see the benefit of all of these rule changes. It seems more convoluted for no benefit.

Reply:
+1 with those concerns. Maybe we can split off the dependabot stuff and take that in?

Reply:
This basically utilizes the core features of Make. Did this contribution to this repo because the other SPIFFE repos have similar approach. This makes it easier for contributors to recognize similarities in the repos and builds, hopefully easing the way people can contribute.

I do understand the argument of added complexity.